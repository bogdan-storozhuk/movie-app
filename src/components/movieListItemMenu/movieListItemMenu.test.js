import React from "react";
import { Provider } from "react-redux";
import { screen, render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import configureMockStore from "redux-mock-store";

import MovieListItemMenu from "./";

const mockStore = configureMockStore();
const store = mockStore({});

describe("MovieListItemMenu", () => {
  const openEditMovieModal = jest.fn();
  const openDeleteMovieModal = jest.fn();
  const id = 2;
  it("should match snapshot with closed Toast Menu", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <MovieListItemMenu
          id={id}
          openEditMovieModal={openEditMovieModal()}
          openDeleteMovieModal={openDeleteMovieModal()}
        />
      </Provider>
    );

    expect(
      asFragment(
        <MovieListItemMenu
          id={id}
          openEditMovieModal={openEditMovieModal()}
          openDeleteMovieModal={openDeleteMovieModal()}
        />
      )
    ).toMatchSnapshot();
  });

  it("should match snapshot with open Toast Menu", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <MovieListItemMenu
          id={id}
          openEditMovieModal={openEditMovieModal()}
          openDeleteMovieModal={openDeleteMovieModal()}
        />
      </Provider>
    );
    const button = screen.queryByTestId("movie-list-item-toggle");
    expect(button).toBeInTheDocument();

    userEvent.click(button);

    expect(
      asFragment(
        <MovieListItemMenu
          id={id}
          openEditMovieModal={openEditMovieModal()}
          openDeleteMovieModal={openDeleteMovieModal()}
        />
      )
    ).toMatchSnapshot();
  });

  it("should correctly handle openEditMovieModal event", () => {
    render(
      <Provider store={store}>
        <MovieListItemMenu
          id={id}
          openEditMovieModal={openEditMovieModal()}
          openDeleteMovieModal={openDeleteMovieModal()}
        />
      </Provider>
    );

    const toggleButton = screen.queryByTestId("movie-list-item-toggle");
    expect(toggleButton).toBeInTheDocument();
    userEvent.click(toggleButton);

    const span = screen.queryByText("Edit");
    expect(span).toBeInTheDocument();
    userEvent.click(span);
    expect(openEditMovieModal).toHaveBeenCalled();
  });

  it("should correctly handle openDeleteMovieModal event", () => {
    render(
      <Provider store={store}>
        <MovieListItemMenu
          id={id}
          openEditMovieModal={openEditMovieModal()}
          openDeleteMovieModal={openDeleteMovieModal()}
        />
      </Provider>
    );

    const toggleButton = screen.queryByTestId("movie-list-item-toggle");
    expect(toggleButton).toBeInTheDocument();
    userEvent.click(toggleButton);

    const span = screen.queryByText("Delete");
    expect(span).toBeInTheDocument();
    userEvent.click(span);
    expect(openDeleteMovieModal).toHaveBeenCalled();
  });
});
