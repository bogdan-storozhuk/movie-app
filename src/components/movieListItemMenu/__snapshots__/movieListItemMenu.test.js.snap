// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieListItemMenu should match snapshot with closed Toast Menu 1`] = `
<DocumentFragment>
  <div
    class="MovieListItemMenu"
  >
    <button
      class="MovieListItemMenu-Toggle"
      data-testid="movie-list-item-toggle"
    >
      <svg
        fill="currentColor"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.5 13a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zm0-5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zm0-5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0z"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;

exports[`MovieListItemMenu should match snapshot with open Toast Menu 1`] = `
<DocumentFragment>
  <div
    class="MovieListItemMenu"
  >
    <button
      class="MovieListItemMenu-Toggle"
      data-testid="movie-list-item-toggle"
    >
      <svg
        fill="currentColor"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.5 13a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zm0-5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zm0-5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0z"
        />
      </svg>
    </button>
    <div
      aria-atomic="true"
      aria-live="assertive"
      class="fade toast show showing"
      role="alert"
    >
      <div
        class="toast-header"
      >
        <button
          aria-label="Close"
          class="btn-close"
          data-dismiss="toast"
          type="button"
        />
      </div>
      <div
        class="toast-body"
      >
        <ul
          class="MovieListItemMenu-Controls"
        >
          <li>
            <span
              class="Controls-Span"
            >
              Edit
            </span>
          </li>
          <li>
            <span
              class="Controls-Span"
            >
              Delete
            </span>
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
